builder/parser/line_parsers.go

```diff
@@ -209,34 +208,28 @@ func parseString(rest string) (*Node, map[string]bool, error) {
 
 // parseJSON converts JSON arrays to an AST.
 func parseJSON(rest string) (*Node, map[string]bool, error) {
-	var (
-		myJson   []interface{}
-		next     = &Node{}
-		orignext = next
-		prevnode = next
-	)
-
+	var myJson []interface{}
```

Comment:
I actually started with that:
```irc
16:59:08 <tianon> heh, we actually get a failing test because the nested arrays are no longer an error, but actually get passed directly to the shell as-is
17:07:08 <tianon> so we _have_ established that if something parses as valid JSON array syntax, it must be valid Dockerfile syntax (and is not passed to the shell)
```